clients/src/main/java/org/apache/kafka/common/network/ChannelBuilders.java

```diff
@@ -159,24 +159,25 @@ private static ChannelBuilder create(SecurityProtocol securityProtocol,
     }
 
     // Visibility for testing
-    protected static Map<String, Object> channelBuilderConfigs(final AbstractConfig config, final ListenerName listenerName) {
-        Map<String, ?> parsedConfigs;
+    @SuppressWarnings("unchecked")
+    static Map<String, Object> channelBuilderConfigs(final AbstractConfig config, final ListenerName listenerName) {
+        Map<String, Object> parsedConfigs;
         if (listenerName == null)
-            parsedConfigs = config.values();
+            parsedConfigs = (Map<String, Object>) config.values();
         else
             parsedConfigs = config.valuesWithPrefixOverride(listenerName.configPrefix());
 
-        // include any custom configs from original configs
-        Map<String, Object> configs = new HashMap<>(parsedConfigs);
         config.originals().entrySet().stream()
             .filter(e -> !parsedConfigs.containsKey(e.getKey())) // exclude already parsed configs
             // exclude already parsed listener prefix configs
             .filter(e -> !(listenerName != null && e.getKey().startsWith(listenerName.configPrefix()) &&
                 parsedConfigs.containsKey(e.getKey().substring(listenerName.configPrefix().length()))))
             // exclude keys like `{mechanism}.some.prop` if "listener.name." prefix is present and key `some.prop` exists in parsed configs.
             .filter(e -> !(listenerName != null && parsedConfigs.containsKey(e.getKey().substring(e.getKey().indexOf('.') + 1))))
-            .forEach(e -> configs.put(e.getKey(), e.getValue()));
-        return configs;
+            .forEach(e -> parsedConfigs.put(e.getKey(), e.getValue()));
+        // The callers may add new elements to return map so we should not wrap it to a immutable map. Otherwise,
+        // the callers have to create a new map to carry more elements and then following Get ops are not recorded.
```

Comment:
(1) "so we should not wrap it to a immutable map": It's kind of weird to have a comment on what we don't do.
(2) to return map  => to returned map

Reply:
Sorry for unclear comment. I will revise this comment and move it to method docs